### PR TITLE
Fixed #28547 -- Doc'd encoded URLs in redirect() function

### PR DESCRIPTION
**Pull request by JakeBar**
https://code.djangoproject.com/ticket/28547

### CODE REVIEW

File: django/shortcuts.py [hunk 1]
```diff
@@ -51,8 +51,10 @@ def redirect(to, *args, permanent=False, **kwargs):
 
         * A URL, which will be used as-is for the redirect location.
 
+    URLs are automatically encoded.
```
Comment (knbk) [inline django/shortcuts.py]: This is only the case if a pattern name is passed in, and the result of `reverse()` is returned, other urls are returned as-is.

**Comment by JakeBar** [comment]
@knbk Let me know what you think

**Comment by timgraham** [comment]
I'm not sure why there would be a need to state this considering that the user doesn't have a chance to modify the URL in that case. @Kangaroux, maybe you can shed some light on your confusion that made you create the ticket?

**Comment by Kangaroux** [comment]
@timgraham Consider the case where I want to redirect to a page and provide GET params:

```python
return redirect(reverse("some_page") + "?val1=%s&val2=%s" % (a, b))
```

It's not clear whether I need to import something like `urllib` to encode the URL, or if django will handle that for me automatically.

**Comment by timgraham** [comment]
It looks like that would fall under "A URL, which will be used as-is for the redirect location." right? To me, "used as-is" implies that no encoding is performed.

**Comment by Kangaroux** [comment]
@timgraham So this is where the confusion lies for me. Because as you said, `as-is` implies no extra processing is done on the URL. However some browsers will handle the URL encoding for you automatically, so at first glance it can be a bit ambiguous regarding whether the encoding is done by django or the browser. I tested this by doing a redirect to an unencoded URL and it worked fine so I assumed it was django handling it.

**Comment by timgraham** [comment]
I guess we could say something like "The URL should be quoted if necessary; for example, by using `urllib.parse.quote_plus()`." however my concern is that this advice seems relevant everywhere a redirect happens (e.g. `RedirectView`, `HttpResponseRedirect`, etc.). Is it unusually important to mention here (or everywhere)?

**Comment by Kangaroux** [comment]
@timgraham I think it would be good to mention it everywhere, but in a way that prevents duplicating the docs. So the edit would go under `redirect` and then you would just refer people to look at how `redirect` works in the docs for `RedirectView`, for example (assuming that isn't already there).

**Comment by timgraham** [comment]
Well, I think most people would think that when building a URL manually, it needs to be encoded. This is more of a general web development thing than a Django thing. To extend things a bit further, should `get_absolute_url()` contain a similar note?... and certainly there are other places where this issue might be relevant. We need to be careful about the signal to noise ratio in the documentation and my feeling is that this isn't a common source of confusion that needs to be noted.

**Comment by timgraham** [comment]
Closing per ticket.

**Pull request closed without merging**